nomos-da/kzgrs-backend/src/encoder.rs

```diff
@@ -35,6 +36,30 @@ pub struct EncodedData {
     pub aggregated_column_proofs: Vec<Proof>,
 }
 
+impl EncodedData {
+    pub fn prepare_data(&self) -> impl Iterator<Item = DaBlob> + '_ {
+        let row_proofs_iter = self.rows_proofs.iter().map(|rp| rp.iter());
+
+        self.extended_data
+            .columns()
+            .zip(&self.column_commitments)
+            .zip(row_proofs_iter)
+            .zip(&self.aggregated_column_proofs)
+            .map(
+                move |(((column, column_commitment), row_proofs), aggregated_column_proof)| {
+                    DaBlob {
+                        column,
+                        column_commitment: *column_commitment,
+                        aggregated_column_commitment: self.aggregated_column_commitment,
+                        aggregated_column_proof: *aggregated_column_proof,
+                        rows_commitments: self.row_commitments.clone(),
+                        rows_proofs: row_proofs.cloned().collect(),
```

Comment:
I'm not sure this is doing what it should. We want all proofs for `column 1..n` meaning that you should iterate the `row_proofs_iter` and call `next` for each row to get them in order. Probably you would need to collect the `row_proofs_iter` so you can iterate over it many times. 
If this is not clear I can give it a try.

Reply:
Yes, I was puzzled about this line in python, I didn't come up with a simple test without creating a verifier instance, maybe you have idea for one?

Reply:
You can verify manually, with the methods from `kzgrs`.